Missions/Missions/Games Tab/GameScene.swift

```diff
@@ -0,0 +1,215 @@
+        verticalLine.strokeColor = SKColor.white
+        verticalLine.lineWidth = 5
+        
+        return verticalLine
+    }()
+        
+    override func didMove(to view: SKView) {
+        physicsWorld.contactDelegate = self
+//        getSceneRangeValues()
+        addObservers()
+        setupScene()
+        addChildren()
+        setupCategoryMasks()
+        setupCollisionMasks()
+        setupContactTestBitMasks()
+    }
+
+}
+
+extension GameScene {
+    
+    func addObservers() {
+        NotificationCenter.default.addObserver(self, selector: #selector(updateDestXY(_:)), name: BallPosition.single.notification, object: nil)
+    }
+    
+    func getSceneRangeValues() {
+        minX = (-self.frame.width / 2) + Radius.single.rawValue
+        maxX = (self.frame.width / 2) - Radius.single.rawValue
+        minY = -self.frame.height / 2 + Radius.single.rawValue
+        maxY = self.frame.height / 2 - Radius.single.rawValue
+    }
+    
+    @objc func updateDestXY(_ notification: Notification) {
+        if let xVector = notification.userInfo?["horizontalConstant"] as? CGFloat, let yVector = notification.userInfo?["verticalConstant"] as? CGFloat {
+            destX = xVector
+            destY = yVector
+        }
+    }
+    
+    func setupCategoryMasks() {
+        ballNode.physicsBody?.categoryBitMask = CategoryMask.ball.rawValue
+        foodNode.physicsBody?.categoryBitMask = CategoryMask.food.rawValue
+    }
+    
+    func setupCollisionMasks() {
+        ballNode.physicsBody?.collisionBitMask = CategoryMask.food.rawValue
+        foodNode.physicsBody?.collisionBitMask = CategoryMask.ball.rawValue
+    }
+    
+    func setupContactTestBitMasks() {
+        ballNode.physicsBody?.contactTestBitMask = CategoryMask.food.rawValue
+        foodNode.physicsBody?.contactTestBitMask = ~(CategoryMask.ball.rawValue)
+    }
+    
+    func setupScene() {
+        self.scaleMode = .resizeFill
+        self.backgroundColor = UIColor(white: 0.9, alpha: 1)
+        self.anchorPoint = CGPoint(x: 0.5, y: 0.5)
+        self.physicsWorld.gravity = CGVector(dx: 0.0, dy: -0.0)
+    }
+    
+    func addChildren() {
+        self.addChild(horizontalLine)
+        self.addChild(verticalLine)
+        self.addChild(ballNode)
+        self.addChild(foodNode)
+    }
+    
+    func clamp(_ value: CGFloat, min: CGFloat, max: CGFloat) -> CGFloat {
+        if value > max {
+            return max
+        }
+        if value < min {
+            return min
+        }
+        return value
+    }
+    
+    override func update(_ currentTime: TimeInterval) {
+        updateBallNodePosition()
+    }
+    
+    
+    
+    func updateBallNodePosition() {
+        let minX = (-self.frame.width / 2) + Radius.single.rawValue
+        let maxX = (self.frame.width / 2) - Radius.single.rawValue
+        let minY = -self.frame.height / 2 + Radius.single.rawValue
+        let maxY = self.frame.height / 2 - Radius.single.rawValue
+//
```

Comment:
optional: get rid of comments wherever they show up. If they are TODO items for another pass, then label it as so.